q2_demux/_transformer.py

```diff
@@ -0,0 +1,39 @@
+import tempfile
+import shutil
+import os.path
+
+import skbio
+from q2_types.per_sample_sequences import FastqGzFormat
+
+from .plugin_setup import plugin
+from ._demux import BarcodeSequenceIterator
+from ._format import EMPMultiplexedDirFmt, EMPMultiplexedSingleEndDirFmt
+
+
+@plugin.register_transformer
+def _1(dirfmt: EMPMultiplexedDirFmt) -> BarcodeSequenceIterator:
+    barcode_generator = skbio.io.read(
+        str(dirfmt.barcodes.view(FastqGzFormat).path), format='fastq',
```

Comment:
You can use `str(dirfmt.barcodes.view(FastqGzFormat))`, no need to use `.path`. Change in other 3 places below too